### DISCUSSION TITLE
Boundary condition in initialize a long relativistic beam

### DISCUSSION THREAD

**Discussion by GivenChen** [SmileiPIC/Smilei#740] [Q&A]
Hi,
When I initialize the relativistic beam's self field, at the transverse boundary, there is a longitudinal field component reflecting back to the center. I am wondering how I can mitigate this effect, except from making a larger window? 
For the boundary condition, I tried silver-muller and PML, they don't really work... 
I also read the documentation, it says it needs a "perfectly absorbing boundary condition", is such boundary condition implemented? or will be implemented in the future?
Thanks for any help!

Best,
Given

**Comment by Z10Frank** [comment]
Hello, 
as far as I know no code provides a perfectly absorbing boundary condition, I should correct it with "perfectly matched layers" (PML). Anyway, you have seen by yourself that although PML provide a better absorption than Silver Muller BC, they cannot damp completely all reflections if the field is too high.

I suggest:
- trying with more damping cells in the PML;
- checking that you are below the CFL condition (PML can reduce the maximum stable timestep but no analytical formula exist to know it), by trying to reduce the timestep;
- using a longer window: if the field is low enough maybe the PML can contain the reflections as usual. 
 
In general it is a good idea to have a long and large window containing the bunch, because its self-field at t=0 is found through a Poisson-like equation (an elliptic equation) and it is easy to mess with the BC of the Poisson solver if the beam is too close from the window borders. This solver is not made for a set-up with continuous injection and ejection of the field from the window borders for example.

**Comment by GivenChen** [comment]
Hi Thank you very much! I am also curious that in smilei code is the relativistic beam field initialized in the pml medium according to pml parameters? I am asking this because I found this [reference](https://doi.org/10.1109/TMTT.2019.2946160) suggesting it could be the reason of spurious reflecting.
I tried increasing pml cells to 100, but it didn't help to mitigate the problem. Right now I am using a large transverse window, about 100cm while my physics happen within 10cm, which is pretty computationally inefficient.

**Reply by Z10Frank** [reply]
Hello, do you have Figures of these reflections?
Unfortunately I don't see an immediate solution, the bunch field will be initialised inside the physical domain, with the PML at the borders if they are selected.

**Reply by GivenChen** [reply]
Oh I am sorry, I attached the wrong link, this should be the [reference](https://info.fusion.ciemat.es/OCS/EPS2015PAP/pdf/P1.206.pdf);
The reflections look like this:
![image](https://github.com/user-attachments/assets/a15f1227-231a-403a-9db4-bb5b910731e5)
The relativistic beam is from 0 to 300cm, with width of 1cm at the center(not shown here),  and the Ex component is reflected from the transverse boundary to the center(the light green color).

**Reply by Z10Frank** [reply]
I don't see how to initialise the field in the PML without risk of reflection, but can you share the namelist? When I have time I can try to make a test.

**Reply by GivenChen** [reply]
Hi, Thank you very much!
This is the namelist:
[boundary.zip](https://github.com/user-attachments/files/17298248/boundary.zip)
this file use silver-muller but I did try pml as well, it doesn't really help somehow...
In the last reference I mentioned, it says the pml version they implemneted is described in this [paper,](https://doi.org/10.1006/jcph.2000.6623) which shows greater reflection reduction than traditional pml. I hope this might help.

**Reply by Z10Frank** [reply]
For info, this is the PML formulation in Smilei: https://arxiv.org/pdf/2409.06287
Also, your case seems cylindrically symmetric, can you try to run it in cylindrical geometry with 1 azimuthal mode?
(a tutorial with this geometry: https://smileipic.github.io/tutorials/advanced_wakefield_electron_beam.html)

**Reply by Z10Frank** [reply]
Is it normal that your bunch is relativistic and moving towards the right border and no moving window is defined? 
The impact of the bunch with the right border would definitely create strong reflections.
